### PR TITLE
[CSL-1885] Add track item detail load

### PR DESCRIPTION
**Pull request by stanlp1**

### CODE REVIEW

**Review by jjl014 — COMMENTED** [review]
Nice job! 

I left a couple comments here and there, but otherwise looking pretty sweet! We'll also want to add an example to the README under the `Instrument Behavioral Events` section.

File: AutocompleteClient/FW/Logic/Worker/ConstructorIO.swift [hunk 1]
```diff
@@ -440,6 +440,27 @@ public class ConstructorIO: CIOSessionManagerDelegate {
...
+        - itemName: The item name
+        - variationID: The id of the variation
+        - sectionName: The name of the section the product is in.
+     
+     ### Usage Example: ###
+     ```
+     constructorIO.trackItemDetailLoad(customerID: "7654321-BA", itemName: "Pencil", variationID: "7654321-BA-738", sectionName: "Products")
+     ```
+     */
+    public func trackItemDetailLoad(customerID: String, itemName: String, variationID: String? = nil, sectionName: String? = nil, completionHandler: TrackingCompletionHandler? = nil) {
```
[the remark this answers is not in the document]
Comment (stanlp1) [reply AutocompleteClient/FW/Logic/Worker/ConstructorIO.swift:458]: @jjl014 Should we update it for BrowseResultLoad as well? 

Also noticed the default for BrowseResultLoad was "Not available" while I have "unknown". Which one should we go with?

File: AutocompleteClientTests/FW/Logic/Worker/ConstructorIOIntegrationTests.swift [hunk 2]
```diff
@@ -215,6 +215,36 @@ class ConstructorIOIntegrationTests: XCTestCase {
...
     }
+    
+    func testItemDetailLoad() {
+        let constructorClient = ConstructorIO(config: ConstructorIOConfig(apiKey: unitTestKey))
+        let expectation = XCTestExpectation(description: "Request 204")
+        constructorClient.trackItemDetailLoad(customerID: "10001", itemName: "item1", completionHandler: { response in
+            let cioError = response.error as? CIOError
+            XCTAssertNil(cioError)
+            expectation.fulfill()
+        })
```
Comment (jjl014) [inline AutocompleteClientTests/FW/Logic/Worker/ConstructorIOIntegrationTests.swift:226]: I think we're missing `self.wait(for: expectation)` at the end here

File: AutocompleteClientTests/FW/Logic/Worker/ConstructorIOIntegrationTests.swift [hunk 3]
```diff
@@ -215,6 +215,36 @@ class ConstructorIOIntegrationTests: XCTestCase {
...
+    }
+    
+    func testItemDetailLoad_WithVariationID() {
+        let constructorClient = ConstructorIO(config: ConstructorIOConfig(apiKey: unitTestKey))
+        let expectation = XCTestExpectation(description: "Request 204")
+        constructorClient.trackItemDetailLoad(customerID: "10001", itemName: "item1", variationID: "20001", completionHandler: { response in
+            let cioError = response.error as? CIOError
+            XCTAssertNil(cioError)
+            expectation.fulfill()
+        })
```
Comment (jjl014) [inline AutocompleteClientTests/FW/Logic/Worker/ConstructorIOIntegrationTests.swift:236]: Same as above, missing `self.wait(for: expectation)`

File: AutocompleteClientTests/FW/Logic/Worker/ConstructorIOIntegrationTests.swift [hunk 4]
```diff
@@ -215,6 +215,36 @@ class ConstructorIOIntegrationTests: XCTestCase {
...
+    }
+    
+    func testItemDetailLoad_WithSection() {
+        let constructorClient = ConstructorIO(config: ConstructorIOConfig(apiKey: unitTestKey))
+        let expectation = XCTestExpectation(description: "Request 204")
+        constructorClient.trackItemDetailLoad(customerID: "10001", itemName: "item1", sectionName: "Merchants", completionHandler: { response in
+            let cioError = response.error as? CIOError
+            XCTAssertNil(cioError)
+            expectation.fulfill()
+        })
```
Comment (jjl014) [inline AutocompleteClientTests/FW/Logic/Worker/ConstructorIOIntegrationTests.swift:246]: Samesies, `self.wait(for: expectation)`

File: AutocompleteClientTests/FW/Logic/Worker/ConstructorIOIntegrationTests.swift [hunk 5]
```diff
@@ -215,6 +215,36 @@ class ConstructorIOIntegrationTests: XCTestCase {
...
+    }
+    
+    func testItemDetailLoad_WithSection() {
+        let constructorClient = ConstructorIO(config: ConstructorIOConfig(apiKey: unitTestKey))
+        let expectation = XCTestExpectation(description: "Request 204")
+        constructorClient.trackItemDetailLoad(customerID: "10001", itemName: "item1", sectionName: "Merchants", completionHandler: { response in
```
Comment (jjl014) [inline AutocompleteClientTests/FW/Logic/Worker/ConstructorIOIntegrationTests.swift:242]: I'm not sure where the "Merchants" section is coming from, but should we be using the default "Products" section here instead?

**Review by jjl014 — COMMENTED** [review]
I couldn't figure out why the `trackPurchase` test kept failing when I ran the whole test suite, but passed when ran individually. It looks like it was because we weren't waiting for the expectations to complete (left some comments around that).

The rest of the changes look good to me though! If we can also update the README (and probably the Wiki as well) on how they can call the event, that would be awesome. Thanks!

**Review by jjl014 — APPROVED** [review]
Nicee. This looks great to me! 🚀

**Pull request merged**